tbx/core/blocks.py

```diff
@@ -581,7 +585,7 @@ def clean(self, value):
                         errors[key].append(message)
                         non_block_errors.append(ValidationError(message))
 
-        if errors:
+        if errors or non_block_errors:
```

Comment:
```suggestion
        if errors:
```

Reply:
It turned out we still needed the `or non_block_errors` here because of the way the button link and text validation had been previously implemented. But I have implemented all your other suggestions - many thanks for your help on this!